R/db-odbc-mssql.R

```diff
@@ -223,11 +209,17 @@ mssql_generic_infix <- function(if_select, if_filter) {
 }
 
 mssql_sql_if <- function(cond, if_true, if_false = NULL) {
+
+  old <- sql_current_context()
+  on.exit(set_current_context(old), add = TRUE)
+  set_current_context(list(clause = ""))
+  cond <- build_sql(cond)
+
   build_sql(
```

Comment:
Just wondering: can we just just call `sql_if()` here?

Reply:
That function lacks the `else` step I had to add to this function so that the first case I posted in the before-and-after comment works.  Didn't want to change `sql_if()` because of I'm not sure if it would mess up anything for the other translations.

Reply:
That doesn't seem right to me — you shouldn't be adding an else block that returns an empty string.

Reply:
Idk what happened, I swear I tested it before replying this morning.  I tried switching to `sql_if()` again to trace the error, but everything worked :) Sorry for the confusion, I'll attribute it to lack of coffee